### PR TITLE
Add the rest of the jinjalint checks documentation

### LINKED ISSUE #39: Documentation for jinjalint checks

### PR DESCRIPTION
**Pull request by mschwager**
Building on #40.

Fixes #39.

### CODE REVIEW

File: docs/jinjalint/missing-doctype.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+**tl;dr** HTML documents must include a [`DOCTYPE` declaration](https://developer.mozilla.org/en-US/docs/Glossary/Doctype)
+at the beginning of the document. This prevents the browser from switching into
+["quirks mode"](https://developer.mozilla.org/en-US/docs/Quirks_Mode_and_Standards_Mode).
+Quirks mode can also impact the security of the web page.
+
+## Description
+
+Web pages missing a `DOCTYPE` declaration may be vulnerable to many different
+esoteric forms of [XSS attacks](https://owasp.org/www-community/attacks/xss/).
+For example, Javascript execution via the [`<title>` tag or via a VML frame](https://html5sec.org/).
```
Comment (minusworld) [inline docs/jinjalint/missing-doctype.md]: Grammar suggestion:

...esoteric forms of [XSS attacks](...),
such as Javascript executions via the ...

File: docs/jinjalint/missing-meta-charset.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+in the `<head>` of the document. This prevents the browser from incorrectly
+interpreting the character encoding of the document. The character encoding can
+impact the security of the web page.
+
+## Description
+
+Web pages missing a `<meta>` `charset` declaration may be vulnerable to many different
+esoteric forms of [XSS attacks](https://owasp.org/www-community/attacks/xss/).
+For example, Javascript execution due to [`CESU-8`, `UTF-7`, `BOCU-1`, or `SCSU`
+encoding](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta).
```
Comment (minusworld) [inline docs/jinjalint/missing-meta-charset.md]: Same suggestion as above:

...esoteric forms of [XSS attacks](https://owasp.org/www-community/attacks/xss/),
such as Javascript execution...

File: docs/jinjalint/missing-meta-content-type.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+against the browser incorrectly interpreting the character encoding of the
+document. The character encoding can impact the security of the web page.
+
+## Description
+
+Web pages missing a `<meta>` `Content-Type` declaration may be vulnerable to many different
+esoteric forms of [XSS attacks](https://owasp.org/www-community/attacks/xss/).
+For example, Javascript execution due to [`CESU-8`, `UTF-7`, `BOCU-1`, or `SCSU`
+encoding](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta).
+
```
Comment (minusworld) [inline docs/jinjalint/missing-meta-content-type.md]: Same suggestion as above:

...esoteric forms of XSS attacks,
such as Javascript execution...

**Comment by mschwager** [comment]
:tada:

**Pull request merged**